server/src/test/java/org/elasticsearch/action/search/CanMatchPreFilterSearchPhaseTests.java

```diff
@@ -442,7 +442,7 @@ public void sendCanMatch(
                 shardsIter,
                 timeProvider,
                 null,
-                true,
+                shardToSkip.size() > shardsIter.size(),
```

Comment:
@dnhatn this is interesting, it is possible for this test to randomly skip ALL shards, but we have this logic that prefers matching against at least one local shard. 

To make this test pass I addressed this, but what do you think? Could this also be related to: https://github.com/elastic/elasticsearch/issues/92058 ???